openwillis/measures/audio/util/transcribe_util.py

```diff
@@ -8,26 +8,138 @@
 
 import json
 import logging
+from openwillis.measures.audio.util import separation_util as sutil
 
 logging.basicConfig(level=logging.INFO)
 logger=logging.getLogger()
 
-def transcribe_audio(s3uri, region_name, job_name, language = 'en-US'):
+def replace_speaker_labels(data, check_lables, speaker_labels):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    Recursively replaces speaker labels in a nested dictionary or list.
+
+    Parameters:
+    ...........
+    data : dict
+        The nested dictionary or list containing speaker labels.
+    check_lables: list
+        Check on input speaker labels
+    speaker_labels: list
+        Expected speaker labels
+
+    Returns:
+    ...........
+    data : dict or list
+        The modified nested dictionary or list with replaced speaker labels.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+
+    if isinstance(data, dict):
+        for key, value in data.items():
+
+            if key == 'speaker_label':
+                if value == check_lables[0]:
+                    data[key] = speaker_labels[0]
+
+                elif value == check_lables[1]:
+                    data[key] = speaker_labels[1]
+
+                else:
```

Comment:
is this needed?

Reply:
As of now, we are separating recordings for only 2 speakers. However, it is possible that we may have more than 2 speakers in a recording. This check is just to handle a scenario in case we have more than 2 speakers